tester/include/game_tester/game_tester.hpp

```diff
@@ -364,9 +383,95 @@ class game_tester : public TESTER {
                             : abi_ser[game_name].binary_to_variant("session_row", data, abi_serializer_max_time);
     }
 
+  private:
+    void handle_action_data(
+            bytes&& action_data,
+            const int event_type,
+            fc::path event_abi
+    ) {
+        switch (event_type) {
+        case events_id::game_started:
+            event_abi /= "game_started.abi";
+            break;
+        case events_id::action_request:
+            event_abi /= "action_request.abi";
+            break;
+        case events_id::signidice_part_1_request:
+            event_abi /= "signidice_part_1_request.abi";
+            break;
+        case events_id::signidice_part_2_request:
+            event_abi /= "signidice_part_2_request.abi";
+            break;
+        case events_id::game_finished:
+            event_abi /= "game_finished.abi";
+            break;
+        case events_id::game_failed:
+            event_abi /= "game_failed.abi";
+            break;
+        case events_id::game_message:
+            event_abi /= "game_message.abi";
+            break;
+        default:
+            BOOST_TEST_FAIL("Can't interpret event type");
+        }
+
+        const abi_def event_abi_def = fc::json::from_file(event_abi).as<abi_def>();
+        abi_serializer abi_ser;
+        abi_ser.set_abi(event_abi_def, abi_serializer_max_time);
+
+        fc::variant event_struct;
+        if (!action_data.empty()) {
+            event_struct = abi_ser.binary_to_variant(
+                "event_data",
+                action_data,
+                abi_serializer_max_time
+            );
+        }
+
+        if (auto it = _events.find(static_cast<events_id>(event_type)); it != _events.end()) {
+            it->second.emplace_back(std::move(event_struct));
+        } else {
+            _events[static_cast<events_id>(event_type)] = {event_struct, };
+        }
+    }
+
+    void handle_transaction_ptr(const transaction_trace_ptr& transaction_trace) {
+        _events.clear();
+
+        const fc::path cpath = fc::canonical(contracts::events_struct::folder());
+        const abi_def events_abi = fc::json::from_file(contracts::platform::events::abi_path()).as<abi_def>();
+
+        abi_serializer abi_ser;
+        abi_ser.set_abi(events_abi, abi_serializer_max_time);
```

Comment:
It's better to move things like that to another function like init_abis and call once from tester constructor, else that stuff invokes for each trx. And same with `handle_action_data`.